src/Libraries/CoreNodeModels/DefineData.cs

```diff
@@ -135,13 +172,31 @@ public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode
         /// <param name="data"></param>
         private void DataBridgeCallback(object data)
         {
-            var inputs = data as ArrayList;
+            if (data == null) return;
 
-            var inputObject = inputs[0];
+            (bool IsValid, bool UpdateList, DataNodeDynamoType InputType) resultData = (ValueTuple<bool, bool, DataNodeDynamoType>)data;
 
-            if (!InPorts[0].IsConnected)
+            if (IsAutoMode && resultData.UpdateList)
             {
-                return;
+                IsList = !IsList;
```

Comment:
@saintentropy Is there any concern that IsList could be changed in the meantime, making inversion the wrong thing to do here? If so, then perhaps we should return a `bool?` and set IsList to match if it's not null. But if nothing can (reasonably) happen in-between the function call and the DataBridgeCallback execution then no need to add complexity, I think.

Reply:
I don't think you mean Manual mode, right? They way I understand the flow is, if the user interacts with any of the inputs, the node will automatically rerun. If in Manual, then the node will evaluate the current configuration of inputs .. ? 
I really don't think anything can come in-between the `IsList` value being changed and the execution of the dataBridge.